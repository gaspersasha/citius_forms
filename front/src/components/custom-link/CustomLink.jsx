import React from 'react';
import PropTypes from 'prop-types';

export const getClassName = (type) => {
  switch (type) {
    case 'button':
      return 'btn';
    case 'button-forward':
      return 'button-forward';
    case 'button-back':
      return 'link--button-back';
    case 'arrow-forward':
      return 'link--arrow-forward';
    case 'arrow-back':
      return 'link--arrow-back';
    case 'arrow-border-forward':
      return 'link--arrow-border-forward';
    default:
      // allow type to be used for className if not defined above
      return type;
  }
};

const CustomLink = ({ type, url, children, ...props }) => {
  const className = getClassName(type);

  if (url) {
    return (
      <a href={url} className={className} {...props}>
        {children}
      </a>
    );
  }

  // if no url is passed render a div instead
  // (this allows us to to conditional wrapping of elements e.g. Card)
  return (
    <div className={className} {...props}>
      {children}
    </div>
  );
};

CustomLink.propTypes = {
  children: PropTypes.any.isRequired,
  url: PropTypes.string,
  type: PropTypes.string,
};

CustomLink.defaultProps = {
  type: '',
  url: '',
};

export default CustomLink;
