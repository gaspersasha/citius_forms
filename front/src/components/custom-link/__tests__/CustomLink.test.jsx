import React from 'react';
import CustomLink from '../CustomLink';

describe('<CustomLink />', () => {
  let wrapper;

  const getWrapper = (props) =>
    shallow(<CustomLink {...props}>Test</CustomLink>);

  it('renders without errors', () => {
    wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('handles both link and block cases', () => {
    wrapper = getWrapper();
    expect(wrapper.html()).toEqual('<div class="">Test</div>');

    wrapper.setProps({ url: 'test' });
    expect(wrapper.html()).toEqual('<a href="test" class="">Test</a>');
  });

  it('defines className in accordance to the type prop', () => {
    wrapper = getWrapper({ type: 'button' });
    expect(wrapper.prop('className')).toEqual('btn');

    wrapper.setProps({ type: 'button-forward' });
    expect(wrapper.prop('className')).toEqual('button-forward');

    wrapper.setProps({ type: 'button-back' });
    expect(wrapper.prop('className')).toEqual('link--button-back');

    wrapper.setProps({ type: 'arrow-forward' });
    expect(wrapper.prop('className')).toEqual('link--arrow-forward');

    wrapper.setProps({ type: 'arrow-back' });
    expect(wrapper.prop('className')).toEqual('link--arrow-back');

    wrapper.setProps({ type: 'arrow-border-forward' });
    expect(wrapper.prop('className')).toEqual('link--arrow-border-forward');
  });
});
