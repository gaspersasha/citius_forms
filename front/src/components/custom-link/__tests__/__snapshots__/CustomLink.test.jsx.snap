// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomLink /> renders without errors 1`] = `
<div
  className=""
>
  Test
</div>
`;
